src/comparator.rs

```diff
@@ -0,0 +1,826 @@
+    feature = "stm32g484"
+))]
+use crate::gpio::gpiob::{PB10, PB11, PB12, PB13, PB15};
+
+#[cfg(any(
+    feature = "stm32g473",
+    feature = "stm32g483",
+    feature = "stm32g474",
+    feature = "stm32g484"
+))]
+use crate::gpio::gpioc::{PC6, PC7, PC8};
+
+#[cfg(any(
+    feature = "stm32g473",
+    feature = "stm32g483",
+    feature = "stm32g474",
+    feature = "stm32g484"
+))]
+use crate::gpio::gpiod::{PD10, PD11, PD12, PD13, PD15};
+
+use crate::gpio::gpioc::{PC0, PC1, PC2};
+use crate::gpio::gpiod::PD14;
+use crate::gpio::gpioe::{PE7, PE8};
+use crate::gpio::gpiof::{PF1, PF4};
+use crate::rcc::{Clocks, Rcc};
+use crate::stm32::{COMP, EXTI};
+
+/// Enabled Comparator (type state)
+pub struct Enabled;
+
+/// Enabled and locked (config is read only)
+pub struct Locked;
+
+/// Disabled Comparator (type state)
+pub struct Disabled;
+
+pub trait ED {}
+impl ED for Enabled {}
+impl ED for Disabled {}
+
+pub trait EnabledState {}
+impl EnabledState for Enabled {}
+impl EnabledState for Locked {}
+
+macro_rules! impl_comp {
+    ($($t:ident: $reg_t:ident, $reg:ident,)+) => {$(
+        pub struct $t {
+            _rb: PhantomData<()>,
+        }
+
+        impl $t {
+            pub fn csr(&self) -> &$crate::stm32::comp::$reg_t {
+                // SAFETY: The COMP1 type is only constructed with logical ownership of
+                // these registers.
+                &unsafe { &*COMP::ptr() }.$reg
+            }
+        }
+    )+};
+}
+
+impl_comp! {
+    COMP1: C1CSR, c1csr,
+    COMP2: C2CSR, c2csr,
+    COMP3: C3CSR, c3csr,
+    COMP4: C4CSR, c4csr,
+}
+#[cfg(any(
+    feature = "stm32g473",
+    feature = "stm32g483",
+    feature = "stm32g474",
+    feature = "stm32g484"
+))]
+impl_comp! {
+    COMP5: C5CSR, c5csr,
+    COMP6: C6CSR, c6csr,
+    COMP7: C7CSR, c7csr,
+}
+
+// TODO: Split COMP in PAC
+
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub struct Config {
+    //power_mode: PowerMode,
+    hysteresis: Hysteresis,
+    inverted: bool,
+    //output_xor: bool,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            hysteresis: Hysteresis::None,
+            inverted: false,
+            //power_mode: PowerMode::HighSpeed,
+            //output_xor: false,
+        }
+    }
+}
+
+impl Config {
+    pub fn hysteresis(mut self, hysteresis: Hysteresis) -> Self {
+        self.hysteresis = hysteresis;
+        self
+    }
+
+    pub fn output_inverted(mut self) -> Self {
+        self.inverted = true;
+        self
+    }
+
+    pub fn output_polarity(mut self, inverted: bool) -> Self {
+        self.inverted = inverted;
+        self
+    }
+
+    /*
+    pub fn power_mode(mut self, power_mode: PowerMode) -> Self {
+        self.power_mode = power_mode;
+        self
+    }*/
+
+    /*
+    /// Sets the output to be Comparator 1 XOR Comparator 2.
+    /// Used to implement window comparator mode.
+    pub fn output_xor(mut self) -> Self {
+        self.output_xor = true;
+        self
+    }*/
```

Comment:
AFAIK this is not supported by stm32g4 processors. Should this be removed?

Reply:
Yes we should. We don't need to store code in comments.